File: insertion-sort/Insertion.py
import time


def insertion(arr):
    i = 1
    for j in range(0, len(arr) - 1):
        current = arr[i]
        while (j >= 0 and current < arr[j]):
            arr[j + 1] = arr[j]
            j -= 1
        arr[j + 1] = current
        i += 1
    return arr


start = time.time()
arr = insertion([5, 21, 44, 33, 18, 10])
end = time.time()
print(arr, "time: " + str((end - start) * 10**3))

File: insertion-sort/Insertion.ts
function insertion(arr: number[]) {
	let current: number;
	let i: number;
	for (let j = 0; j < arr.length - 1; j++) {
		i = j + 1;
		current = arr[i];
		while (i > 0 && current < arr[i - 1]) {
			arr[i] = arr[i - 1];
			i--;
		}
		arr[i] = current;
	}
	return arr;
}

const start = performance.now();
const arr = insertion([5, 21, 44, 33, 18, 10]);
const end = performance.now();
console.log(`[${arr}] time: ${end - start}`);
